cime/config/e3sm/machines/config_compilers.xml

```diff
@@ -1791,6 +1791,9 @@ ntel/x86_64/2013/composer_xe_2013/composer_xe_2013_sp1.3.174/mkl/include </base>
   <NETCDF_C_PATH>$ENV{NETCDF_C_PATH}</NETCDF_C_PATH>
   <NETCDF_FORTRAN_PATH>$ENV{NETCDF_FORTRAN_PATH}</NETCDF_FORTRAN_PATH>
   <SUPPORTS_CXX>TRUE</SUPPORTS_CXX>
+  <FC_AUTO_R8>
+    <base> -qrealsize=8 </base>
+  </FC_AUTO_R8>
```

Comment:
Is this change on E3SM master? Why did you need this change?

Reply:
> Is this change on E3SM master? Why did you need this change?

add -qrealsize=8 promotes the default byte size for REAL variables declared without an explicit byte size which will improve the robustness of the code.

Reply:
We recommend against auto-promotion of variables in E3SM.

Reply:
We recommend against auto-promotion of variables in E3SM.

Reply:
> We recommend against auto-promotion of variables in E3SM.

sorry, if you check both the e3sm and acme config_compilers.xml, there are r8 auto-promotion option activated for many compilers.

Reply:
On E3SM master, realsize only appears in one place: the generic ibm config compilers entry.

Our goal is to have all fortran variables and constants declare their precision explicitly so -qrealisize should have no effect/not be needed.

Reply:
> On E3SM master, realsize only appears in one place: the generic ibm config compilers entry.
> 
> Our goal is to have all fortran variables and constants declare their precision explicitly so -qrealisize should have no effect/not be needed.

Nice comments, this is much better than auto-protomotion options,  I am going to close this PR, and create a new one without this option. Thanks very much.